fix(Avatar): fixing issues for the height of the element and the initials generation

mnajdova opened the pull request:
# Avatar

This PR addresses the issues created regarding the avatar components (https://github.com/stardust-ui/react/issues/31 and https://github.com/stardust-ui/react/issues/32)

## Issue #31: Avatar status icon makes the whole component taller
Improved the style of the Avatar component so that the height would be consistent when using the presence icon.

![image](https://user-images.githubusercontent.com/4512430/43531611-6c9556b8-95b0-11e8-9ffa-df4914a2efe2.png)

## Issue #32: Avatar initials parsing should take long / special names into account
Improved default generation of the intials by removing the content inside brackets: [], (), {} and showing just the initials of the first and last word of the name.

Other then this, there is additional property added: `generateInitials` for custom initial generation.

## Fixing images in the avatar examples
Changed the path to the images to correspond accordingly with the images inside the Image component example

## [SUGGESTION NEEDED]
There was strange need for changing the style for the avatar with size 1, containing image and presence icon. If anyone can suggest different approach or some other change, I would appreciate.
```
const getPresenceSpanTop = (size: number, presenceIconPadding: number, src: string) => {
  // TODO check why we need this ?!
  if (src && size === 1) {
    return getPresenceIconSize(size) * 1.5 + getPresenceIconPadding(size, presenceIconPadding) * 2
  }
  return getPresenceIconSize(size) + getPresenceIconPadding(size, presenceIconPadding)
}
````

mnajdova commented:
Maybe it would be a good idea the generate command to create the test for the component as well? The structure for the tests is the same as the components and the naming convention so far is _ComponentName-test.ts_. In the test we could provide the isConformance test by default.

kuzhelov commented:
yes, sure, we should consider that - however, am rather inclined to address this problem as a separate PR, and see this one as a remedy for developer to be able to quickly set up docs as a dev environment for new component :) @mnajdova, what do you think?

mnajdova commented:
Sure, make sense... :)

levithomason commented:
What about keeping a directory of template files and replacing their contents?  This would make reviewing and changing the templates very simple.

We could also very easily copy the files, update their contents and place them in a destination directory using the existing gulp infrastructure.  This would remove the need for the touch, ensureDir, and createFileTree functions since gulp does this automatically. Pseudo example:

```jsx
// TemplateComponent.tsx

const $COMPONENT = () => <$ELEMENT />
```

```jsx
// gulp/tasks/generate.ts

task('generate:component', () => {
  const { name, element } = require('yargs').argv

  return src('./TemplateComponent.tsx')
    .pipe(g.replace('$COMPONENT', name)
    .pipe(g.replace('$ELEMENT', element)
    .pipe(dest(paths.src('components'))
})
```

```jsx
// package.json
{
  scripts: {
    "generate:component": "gulp generate:component"
  }
}
```

Then we can use it with command line args like so:

```bash
$ yarn generate:component Button
```

This could also be easily extended to take a type and name, etc:
```bash
$ yarn generate <type> <name>
```

levithomason commented:
This looks good for now.  Merging to try an catch up on PRs 👍

Commit:
-improved logic for generating initials of the avatar
-added prop generateInitials for custom logic for generating the initials
-fixed style to be consistent when the presence icon is shown
-fixed images in the avatar examples

Commit:
-added tests for the generateInitials methods im the Avatar component

Commit:
-refactored generateInitials method and added as a default prop
-changed the names in the avatarRules, in order to avoid using specific tag names

Commit:
-fixed Avatar-test.tsx

Commit:
Merge branch 'master' into fix/avatar-issues

Commit:
-added records in the changelog

Commit:
-removed empty line

Commit:
Merge branch 'master' into fix/avatar-issues

# Conflicts:
#	CHANGELOG.md

Commit:
-replaced Avatar presence content and wrapper with circular icon
-added borderSize variable and micro side for the Icon component

Commit:
-removed borderSize from the icon component

Commit:
-removed inline style in the Avatar size example

Commit:
-implementing the background of the padding as inherited

Commit:
-replaced generateInitials with getInitials
-removed inline styles from the examples
-added excluded initials example

Commit:
-added presence indicator background variable with default to white
-added example for showing how can this background be altered with different color

Commit:
Merge branch 'master' into fix/avatar-issues

Commit:
-updated changelog
